src/request/hackney_client.erl

```diff
@@ -0,0 +1,20 @@
+-module(hackney_client).
+
+-behavior(http_client).
+
+-export([request/4]).
+
+request(Method, Url, ReqBody, ReqHeaders) ->
+  case hackney:request(Method, Url, ReqHeaders, ReqBody, []) of
+    {ok, StatusCode, RespHeaders, ClientRef} ->
+      if
+        StatusCode < 300 ->
+          {ok, Body} = hackney:body(ClientRef),
+          {ok, [{status_code, StatusCode}, {headers, RespHeaders}, {body, Body }]};
```

Comment:
It is probably a better idea to return a map here instead of the proplist.
An equivalent would be:
```Erlang
#{status_code => StatusCode, headers => Headers, body => Body}
```

The proplists were used widely for that kind of data structure when there was no map in Erlang yet.
But maps play much more nicely with pattern matching, and Elixir relies on them heavily.
It would be great to use the map in the error case too!

Reply:
Updated this as well - im still not sure its recognizing the behavior I defined